instrumentation/ruby_kafka/lib/opentelemetry/instrumentation/ruby_kafka/events/connection/request.rb

```diff
@@ -0,0 +1,40 @@
+# frozen_string_literal: true
+
+# Copyright 2020 OpenTelemetry Authors
+#
+# SPDX-License-Identifier: Apache-2.0
+
+module OpenTelemetry
+  module Instrumentation
+    module RubyKafka
+      module Events
+        module Connection
+          # The Request module contains the instrumentation for connection request event
+          module Request
+            extend self
+
+            EVENT_NAME = 'request.connection.kafka'
+            SPAN_NAME = 'kafka.connection.request'
+
+            def subscribe!
+              ::ActiveSupport::Notifications.subscribe(EVENT_NAME) do |_name, start, finish, _id, payload|
```

Comment:
There's a lot of duplicate code ahead for all of the notifications, this was intentional as I'm not overly familiar with kafka and wanted to keep things as simple as possible.  Open to feedback on we might want to clean this up a bit.

Reply:
I generally am fine with a lot of this repetitive code, I think eventually the goal here should be to abstract things into a general active support base instrumentation module, which is the approach a few vendors have taken. at this point though i don't think it's really worth that much to invest in determining that pattern, so the repetition is fine with me